api/workOrder/workOrderRouter.js

```diff
@@ -0,0 +1,494 @@
+ *    tags:
+ *      - workOrder
+ *    parameters:
+ *      - $ref: '#/components/parameters/workOrderId'
+ *    responses:
+ *      200:
+ *        description: A workOrder object
+ *        content:
+ *          application/json:
+ *            schema:
+ *              $ref: '#/components/schemas/WorkOrder'
+ *      401:
+ *        $ref: '#/components/responses/UnauthorizedError'
+ *      404:
+ *        description: 'WorkOrder not found'
+ */
+router.get(
+  [
+    '/:id',
+    '/users?/workOrders?/:id',
+    '/users?/orders?/:id',
+    '/users?/:userId/workOrders?/:id',
+    '/users?/:userId/orders?/:id',
+    '/company/workOrders?/:id',
+    '/company/orders?/:id',
+    '/company/:companyId/workOrders?/:id',
+    '/company/:companyId/orders?/:id',
+    '/company/users?/:userId/workOrders?/:id',
+    '/company/users?/:userId/orders?/:id',
+    '/company/:companyId/users?/:userId/workOrders?/:id',
+    '/company/:companyId/users?/:userId/orders?/:id',
+    '/property/workOrders?/:id',
+    '/property/orders?/:id',
+    '/property/:propertyId/workOrders?/:id',
+    '/property/:propertyId/orders?/:id',
+    '/company/property/:propertyId/workOrders?/:id',
+    '/company/property/:propertyId/orders?/:id',
+    '/company/:companyId/property/:propertyId/workOrders?/:id',
+    '/company/:companyId/property/:propertyId/orders?/:id',
+  ],
+  authRequired,
+  function (req, res) {
+    const id = String(req.params.id);
+    WorkOrders.findById(id)
+      .then((workOrder) => {
+        if (workOrder) {
+          res.status(200).json(workOrder);
+        } else {
+          res.status(404).json({ error: 'WorkOrderNotFound' });
+        }
+      })
+      .catch((err) => {
+        res.status(500).json({ error: err.message });
+      });
+  }
+);
+
+router.get(
+  [
+    '/company/:companyId',
+    '/company/:companyId/workOrders?',
+    '/company/:companyId/orders?',
+  ],
+  function (req, res) {
+    const companyId = req.params.companyId;
+    console.log(companyId);
+    WorkOrders.findBy({ company: companyId })
+      .then((workOrder) => {
+        if (workOrder) {
+          res.status(200).json(workOrder);
+        } else {
+          res.status(404).json({ error: 'WorkOrderNotFound' });
+        }
+      })
+      .catch((err) => {
+        res.status(500).json({ error: err.message });
+      });
+  }
+);
+
+router.get(
+  [
+    '/property/:propertyId',
+    '/property/:propertyId/workOrders?',
+    '/property/:propertyId/orders?',
+    '/company/property/:propertyId',
+    '/company/property/:propertyId/workOrders?',
+    '/company/property/:propertyId/orders?',
+    '/company/:companyId/property/:propertyId',
+    '/company/:companyId/property/:propertyId/workOrders?',
+    '/company/:companyId/property/:propertyId/orders?',
+  ],
+  function (req, res) {
+    const propertyId = req.params.propertyId;
+    console.log(propertyId);
+    WorkOrders.findBy({ property: propertyId })
+      .then((workOrder) => {
+        if (workOrder) {
```

Comment:
Since this returns an array you will probably want to change your condition to `if(workOrder.length > 0)` as it is returning a 200 with an empty array for nonexisting properties/properties with no work orders

Reply:
Yes. That is better practice.